src/main/java/com/teragrep/cfe_16/EventManager.java

```diff
@@ -142,30 +148,26 @@ public ObjectNode convertData(
          * After the event is handled, it is assigned as a value to previousEvent
          * variable.
          */
-        HttpEventData eventData = null;
-        Converter converter = new Converter(headerInfo);
-        List<SyslogMessage> syslogMessages = new ArrayList<SyslogMessage>();
+        HttpEventData eventData = new HttpEventData();
+        final Converter converter = new Converter(headerInfo);
+        final List<SyslogMessage> syslogMessages = new ArrayList<>();
         while (parser.hasNext()) {
-            previousEvent = eventData;
-            String jsonObjectStr = parser.next().toString();
-            eventData = verifyJsonData(jsonObjectStr, previousEvent);
-            eventData = assignMetaData(eventData, authToken, channel);
-            SyslogMessage syslogMessage = converter.httpToSyslog(eventData);
-            syslogMessages.add(syslogMessage);
-        }
+            try {
+                previousEvent = eventData;
+                final JsonNode jsonNode = new ObjectMapper().readTree(parser.next().toString());
+                final JsonEvent jsonEvent = new JsonEventImpl(jsonNode);
+                eventData.setEvent(jsonEvent.asEvent());
+                eventData = handleTime(eventData, jsonNode, previousEvent);
+                eventData = assignMetaData(eventData, authToken, channel);
 
-        /*
-         * SyslogMessage syslogMessage =
-         * converter.getHeaderInfoSyslogMessage(headerInfo);
-         * requestInfo.getConvertedData().add(syslogMessage);
-         */
-        /*
-         * After all the events are sent, previousEvent object is set to null, the
-         * events are sent with the Acknowledgements and ack id and JSON node with an ack id
-         * will be returned informing that the sending of the events has been
-         * successful.
-         */
-        previousEvent = null;
+                syslogMessages.add(converter.httpToSyslog(eventData));
```

Comment:
please do not nest multiple method calls. this will cause stacktraces to be hard to read as it only says exception at L163 and not that if .add or .httpToSyslog threw. @q22u please add instructions to documentation about not to nest method calls.